bundles/framework/myplaces3/CategoryHandler.js

```diff
@@ -0,0 +1,700 @@
+                me.instance.sandbox.postRequestByName('EnableMapKeyboardMovementRequest');
+            });
+            var cancelBtn = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            cancelBtn.setTitle(me.loc('buttons.cancel'));
+            cancelBtn.setHandler(function () {
+                dialog.close();
+                me.instance.sandbox.postRequestByName('EnableMapKeyboardMovementRequest');
+            });
+            buttons.push(cancelBtn);
+            buttons.push(saveBtn);
+            dialog.makeModal();
+            dialog.show(me.loc('categoryform.edit.title'), content, buttons);
+            dialog.moveTo('div.personaldata ul li select', 'right');
+            //bind listeners etc. for category form
+            form.start();
+
+        },
+        showValidationErrorMessage: function (errors) {
+            var dialog = Oskari.clazz.create('Oskari.userinterface.component.Popup'),
+                okBtn = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            okBtn.setTitle(this.loc('buttons.ok'));
+            okBtn.addClass('primary');
+            okBtn.setHandler(function () {
+                dialog.close(true);
+            });
+            var content = jQuery('<ul></ul>'),
+                i,
+                row;
+            for (i = 0; i < errors.length; i += 1) {
+                row = jQuery('<li></li>');
+                row.append(errors[i].error);
+                content.append(row);
+            }
+            dialog.makeModal();
+            dialog.show(this.loc('validation.title'), content, [okBtn]);
+
+        },
+        /**
+         * @method _showMessage
+         * Shows user a message with ok button
+         * @private
+         * @param {String} title popup title
+         * @param {String} message popup message
+         */
+        _showMessage: function (title, message) {
+            var dialog = Oskari.clazz.create('Oskari.userinterface.component.Popup'),
+                okBtn = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            okBtn.setTitle(this.loc('buttons.ok'));
+            okBtn.addClass('primary');
+
+            okBtn.setHandler(function () {
+                dialog.close(true);
+            });
+            dialog.show(title, message, [okBtn]);
+        },
+
+        /**
+         * @method hasIllegalChars
+         * Checks value for problematic characters
+         * @return {Boolean} true if value has illegal characters
+         */
+        hasIllegalChars: function (value) {
+            this.validateTool.setValue(value);
+            return !this.validateTool.checkValue();
+        },
+
+        /**
+         * @method _validateNumber
+         * Checks value for number and number range
+         * @return {Boolean} true if value is ok
+         * @private
+         */
+        _validateNumber: function (value, min, max) {
+            return this.validateTool.validateNumberRange(value, min, max);
```

Comment:
These validate functions could be moved to Oskari.util: https://github.com/oskariorg/oskari-frontend/blob/develop/src/util.js or even create some new Oskari.validate.isInRange(value, min, max)